syntax = "proto3";

package auth.v1;

option go_package = "auth/v1;authv1";

import "user/v1/user.proto";

service AuthService {
  rpc SignUp(SignUpRequest) returns (SignUpResponse);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
}

message SignUpRequest {
  string username = 1;
  string password = 2;
}

message SignUpResponse {
  user.v1.User user = 1;
  string token = 2;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  user.v1.User user = 1;
  string token = 2;
}

message ValidateTokenRequest {
  string token = 1;
}

message ValidateTokenResponse {
  bool valid = 1;
  user.v1.User user = 2;
}